File: SampleAppObjC/sources/SDKDelegate.m
//
//  SDKDelegate.m
//  SampleAppObjC
//
//  Copyright © 2017 BEAR SAS. All rights reserved.
//

#import "SDKDelegate.h"
#import "Cache.h"

@implementation SDKDelegate

- (void)assetClickedWith:(NSInteger)assetId {
    NSLog(@"assetTappedWithAssetId = %li", (long)assetId);
}

- (void)markerNotRecognized {
    NSLog(@"markerNotRecognized");
}

- (void)recognizedMarkerWithId:(NSInteger)markerId assetIds:(NSArray<NSNumber *> * _Nonnull)assetIds {
    NSLog(@"recognizedMarkerWithId = %ld, assetIds = %@", (long)markerId, assetIds);
    [Cache cacheRecognizedMarkerId:markerId];
}

- (void)scannerStateChanged:(enum BearScannerState)state {
    switch (state) {
        case BearScannerStateIdle:
            NSLog(@"BearScannerStateIdle");
            break;
        case BearScannerStateScanning:
            NSLog(@"BearScanStateScanning");
            break;
        case BearScannerStateProcessing:
            NSLog(@"BearScanStateProcessing");
            break;
        case BearScannerStateRendering:
            NSLog(@"BearScannerStateRendering");
            break;
    }
}

-(void)reachabilityStatusChanged:(BOOL)reachable {
    NSLog(@"reachabilityStatusChanged: reachable = %d", reachable);
}

- (void)didFailWithError:(NSError * _Nonnull)error {
    NSLog(@"didFailWithError %@", error.localizedDescription);
}

@end

File: SampleApp/sources/SDKDelegate.swift
//
//  SDKDelegate.swift
//  SampleApp
//
//  Copyright © 2016-2018 BEAR SAS. All rights reserved.
//

import Foundation
import BearSDK

@objc final class SDKDelegate: NSObject, BearDelegate {

    func markerNotRecognized() {
        debugPrint("called \(#function)")
    }
    
    func scannerStateChanged(_ state: BearScannerState) {
        debugPrint("scanStateChanged \(state)")
    }
    
    func recognizedMarker(withId markerId: Int, assetIds: [Int]) {
        debugPrint("callback \(markerId), assets ids \(assetIds)")
        cacheRecognized(markerId: markerId)
    }
    
    func assetClicked(with assetId: Int) {
        debugPrint("assetClicked with id = \(assetId)")
    }
    
    func didFail(withError error: BearError) {
        debugPrint("called \(#function)")
    }
}
